### DISCUSSION TITLE
Text column with clickable icon

### DISCUSSION THREAD

**Discussion by amritadhikari** [filamentphp/filament#12090] [Help]
### Package

Table builder

### Package Version

v3.x

### How can we help you?

I have built a table and data are all good, now I have to add icon to a column status displayed as ( Pending [icon]), now I want click on the icon(not the text) to send an email. How can I achieve this?
![image](https://github.com/filamentphp/filament/assets/33679299/cb2a424e-919d-4170-8a21-95e84947ed29)

**Comment by rickt96** [comment]
I'm trying to implement a similar function too.
Natively, Filament does not seem to support clickable icons, but it could be solved by implementing a custom column with livewire
[](https://filamentphp.com/docs/3.x/tables/columns/custom)

Edit: On a column you can implement an action, just like this:
```
Tables\Columns\TextColumn::make('condition')
      ->badge()
      ->label('Condition')
      ->sortable()
      ->action(function($state, $record){
          // Send email logic
      }),
```
